src/main/java/model/member/state/implementations/manipulating/ManaDrainMemberState.java

```diff
@@ -2,21 +2,21 @@
 
 import javafx.beans.property.ReadOnlyStringProperty;
 import manager.LanguageUtility;
-import model.member.BattleMember;
+import model.member.interfaces.IBattleMember;
 import model.member.state.MemberStateIcon;
 import model.member.state.RandomPowerMemberState;
 import model.member.state.interfaces.IManipulatingMemberState;
 import model.member.state.interfaces.IRandomMemberState;
 
 public class ManaDrainMemberState extends RandomPowerMemberState implements IRandomMemberState, IManipulatingMemberState {
 
-    public ManaDrainMemberState(String name, int duration, boolean activeRounder, BattleMember source, float maxPower, boolean random) {
+    public ManaDrainMemberState(String name, int duration, boolean activeRounder, IBattleMember source, float maxPower, boolean random) {
         super(name, MemberStateIcon.MANA_DRAIN, duration, activeRounder, source, maxPower, random);
     }
 
     @Override
-    public void apply(BattleMember member) {
-        member.setMana(member.getMana() - Math.round(getEffectPower()));
+    public void apply(IBattleMember member) {
+        member.decreaseMana(member.getMana() - Math.round(getEffectPower()), this.source);
```

Comment:
```suggestion
        member.decreaseMana(Math.round(getEffectPower()), this.source);
```
You dont want to decrease the mana with his current mana